### PR TITLE
Fix UE play settings synchronization

### PR DESCRIPTION
**Pull request by LeoMash**

### CODE REVIEW

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt [hunk 1]
```diff
@@ -8,13 +8,15 @@ import com.jetbrains.rider.plugins.unreal.UnrealHost
 
 class PlaySettings : DefaultActionGroup() {
     override fun update(e: AnActionEvent) {
+        super.update(e)
```
Comment (DecoyRS) [inline src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:11]: update for `DefaultActionGroup` does nothing, it was skipped on purpose

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt [hunk 2]
```diff
@@ -8,13 +8,15 @@ import com.jetbrains.rider.plugins.unreal.UnrealHost
...
         e.presentation.isVisible = e.getHost()?.isUnrealEngineSolution ?: false
         e.presentation.isEnabled = e.getHost()?.isConnectedToUnrealEditor ?: false
```
Comment (DecoyRS) [inline src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:13]: I wonder if this should be visible and enabled for connected version only

File: src/cpp/RiderLink/Source/RiderGameControlExtension/Private/RiderGameControlExtension.cpp [hunk 3]
```diff
@@ -208,31 +198,83 @@ void RequestPlaySession(const FVector* StartLocation, const FRotator* StartRotat
...
+    PlayInSettings->SetPlayNetDedicated(settings.bNetDedicated);
+#else
+    PlayInSettings->bLaunchSeparateServer = settings.bNetDedicated;
+#endif
+    PlayInSettings->LastExecutedPlayModeLocation =
+        settings.bSpawnAtPlayerStart
+            ? PlayLocation_DefaultPlayerStart
+            : PlayLocation_CurrentCameraLocation;
+    PlayInSettings->LastExecutedPlayModeType = settings.PlayMode;
+}
```
Comment (DecoyRS) [inline src/cpp/RiderLink/Source/RiderGameControlExtension/Private/RiderGameControlExtension.cpp:261]: Don't we need to call for `PostEditChange` and `SaveConfig` on `PlayInSettings` after changing them?

**Review by DecoyRS — CHANGES_REQUESTED** [review]
Need to clarify few things

File: src/cpp/RiderLink/Source/RiderGameControlExtension/Private/RiderGameControlExtension.cpp [hunk 4]
```diff
@@ -366,17 +402,23 @@ void FRiderGameControlExtensionModule::StartupModule()
 
             GUnrealEd->PlayWorld->bDebugFrameStepExecution = true;
             GUnrealEd->PlayWorld->bDebugPauseExecution = false;
+		    GUnrealEd->PlaySessionSingleStepped();
```
Comment (DecoyRS) [inline src/cpp/RiderLink/Source/RiderGameControlExtension/Private/RiderGameControlExtension.cpp:405]: formatting bugged out

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt (hunk 1 above)
Comment (LeoMash) [reply src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:11]: Discussed in private about conservative approach

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt (hunk 2 above)
Comment (LeoMash) [reply src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt:13]: IMHO, it looks ok for now and we got used to current UX

**Review by DecoyRS — APPROVED** [review]
LGTM

**Pull request merged**